### PR TITLE
Interdc fix

### PR DESCRIPTION
**Pull request by deepthidevaki**
A failure in sending heartbeats could result in a circular dependencies of metadata among DCs. This pull request has the fix for it and test to cover failure scenario.

### CODE REVIEW

File: src/inter_dc_repl_vnode.erl [hunk 1]
```diff
@@ -91,8 +91,7 @@ handle_command(trigger, _Sender, State=#state{partition=Partition,
                 [_H|_T] ->
                     case inter_dc_communication_sender:propagate_sync(
                            {replicate, Transactions}, DCs) of
-                        ok -> NewReaderState;
-                        _  -> Reader
+                        ok -> NewReaderState
```
Comment (cmeiklejohn) [inline src/inter_dc_repl_vnode.erl]: Could the missing `_` case here generate a `case_clause` error and crash sometimes?

**Comment by cmeiklejohn** [comment]
:+1:, seems good.

My only concern is about the non-exhaustive case clause I mentioned above, that could trigger a crash.

**Comment by deepthidevaki** [comment]
the function returns always 'ok'. Dialyzer was issuing a warning with the additional clause '_'.

**Comment by bieniusa** [comment]
Indeed, the error case can not happen anymore.

**Comment by cmeiklejohn** [comment]
Not for nothing, but why have a `case` statement if it can only return one possible value?

**Pull request merged**